#173 - Static backend cleanup and setup.sh (#176)
blumilksoftware/internships: readme.md

# internships
Website for apprenticeship programme.
# Architecture

> Project contains modified default configuration and is structured as follows:

Reviewer: ```suggestion
The project contains a modified default configuration and is structured as follows:
```

> * `backend` stores configuration for php composer and Laravel application and other php related stuff;

Reviewer: ```suggestion
* `backend` stores configuration for Composer and Laravel application and other PHP related stuff;
```

> * `frontend` is for node's npm, vite, vue and lint configuration - it also contains assets used in webpage;

Reviewer: ```suggestion
* `frontend` is for Node.js npm, Vite, Vue.js and lint configuration - it also contains assets used in the webpage;
```

## Setup and Run
### 1. Run setup script:

> The script will ask you to input a MapBox public key, which will be written into an .env file as VITE_MAPBOX_TOKEN:

Reviewer: ```suggestion
The script will ask you to input a MapBox public key, which will be written into a .env file as VITE_MAPBOX_TOKEN:
```

```shell script
sh setup.sh
```
Or you can do provided steps in the file manually.
Remember to add VITE_MAPBOX_TOKEN key in .env.

### 2. Run development environment:

> To develop application in local environment:

Reviewer: ```suggestion
To develop the application in a local environment:
```

```shell script
docker-compose exec node npm run dev
```

> Application should be available under `localhost:80` or other port if you changed `EXTERNAL_WEBSERVER_PORT` value in `.env` file.

Reviewer: ```suggestion
The application should be available under `localhost:80` or another port if you changed `EXTERNAL_WEBSERVER_PORT` value in the `.env` file.
```

## Development
### Codestyle

> Before committing changes, it is recommended to run codestyle and lint checkers:

Reviewer: ```suggestion
Before committing changes, it is recommended to run code style and lint checkers:
```

```shell script
docker-compose exec php composer run csf
docker-compose exec node npm run lintf
```

> You can remove `f` at the end of these commands, if you don't want to fix issues.

Reviewer: I think that writing commands will be better:

```
docker-compose exec php composer run cs
docker-compose exec node npm run lint
```

Author: Done

> You can remove `f` at the end of these commands, if you don't want to fix issues.

Reviewer: ```suggestion
You can remove `f` at the end of these commands if you don't want to fix issues.
```

# internships
Website for apprenticeship programme.
# Architecture
The project contains a modified default configuration and is structured as follows:
* `backend` stores configuration for Composer and Laravel application and other PHP related stuff;
* `frontend` is for Node.js npm, Vite, Vue.js and lint configuration - it also contains assets used in the webpage;

## Setup and Run
### 1. Run setup script:

> The script will ask you to input a MapBox public key, which will be written into a .env file as VITE_MAPBOX_TOKEN:

Reviewer: Now script isn't asking - it's just friendly remainder on the end to after setting variables in .env file to restart containers.

> The script will ask you to input a MapBox public key, which will be written into a .env file as VITE_MAPBOX_TOKEN:

Author: ```suggestion

```

```shell script

> bash setup.sh

Reviewer: Can we use `sh`?

Author: Yes. 
It was bash before, because `sh` didn't understand double `[` in if statements and that's required in regex. It's gone now, so we can do that.

There's one inconsistency though - the setup file has `#!/bin/bash` at the top, but `sh` forces it to be `#!/bin/sh` if I understood correctly.

Should I update that line?

Reviewer: Yes.

> bash setup.sh

Author: ```suggestion
sh setup.sh
```
